sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -164,4 +164,11 @@ case class HiveTableScanExec(
   }
 
   override def output: Seq[Attribute] = attributes
+
+  override def sameResult(plan: SparkPlan): Boolean = plan match {
```

Comment:
`ReuseExchange` work in parquet/orc format, because `FileSourceScanExec` has override the `sameResult`.
if `left.cleanArgs == right.cleanArgs` return false, we can't run the next `(left.children, right.children).zipped.forall(_ sameResult _)`